crates/sparrow-main/tests/e2e/collection_tests.rs

```diff
@@ -109,6 +108,42 @@ async fn test_bool_to_s_get_static_key() {
     "###);
 }
 
+#[tokio::test]
+async fn test_first_last_map() {
+    // The csv writer does not support map types currently, so the output has been verified
```

Comment:
What if we allowed running to JSON? We could add that as a fast follow-up pretty easily (I have some changes that may get me deep in the testing code, and may be able to pull that in as well).

Reply:
That would be nice -- as discussed, will leave this for a follow-up if it fits into your testing changes.